invenio_analytics_importer/retrieve.py

```diff
@@ -64,50 +66,89 @@ async def get_analytics_for_day(self, method, day):
         if response.text == "No data available":
             return []
 
-        results = response.json()
+        return response.json()
 
-        return results
+    async def fetch_downloads_for_day(self, day):
+        """Fetch downloads analytics for day."""
+        return await self.get_analytics_for_day("Actions.getDownloads", day)
 
 
-async def get_analytics_per_day(api_client, action, days):
-    """Batch retrieve an action analytics per day as dict."""
-    analytics = await asyncio.gather(
-        *[
-            api_client.get_analytics_for_day(action, day)
-            for day in days
-        ]
-    )
-    return dict(zip(days, analytics))
+class DownloadsFetcher:
+    """Fetches downloads."""
 
+    def __init__(self, client):
+        """Constructor."""
+        self.client = client
 
-async def get_downloads_per_day(api_client, days):
-    """Batch retrieve downloads per day as dict."""
-    return await get_analytics_per_day(api_client, "Actions.getDownloads", days)  # noqa
+    async def fetch_analytics_for_day(self, day):
+        """Fetch download analytics for given day."""
+        return await self.client.fetch_downloads_for_day(day)
```

Comment:
This class pattern pushes separation of responsibility pretty far. Don't if commensurate with our reality but good to try it out.